src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -118,13 +119,10 @@ public static AssignmentName parseAssignmentName(String assignmentName) throws P
     public static int parseMaxScore(String maxScore) throws ParseException {
         requireNonNull(maxScore);
         String trimmedMaxScore = maxScore.trim();
-        int parsedScore;
-        try {
-            parsedScore = Integer.parseInt(trimmedMaxScore);
-        } catch (NumberFormatException e) {
-            throw new ParseException("The score must be an integer!");
+        if (!StringUtil.isUnsignedInteger(trimmedMaxScore)) {
+            throw new ParseException(Assignment.MAX_SCORE_MESSAGE_CONSTRAINTS);
         }
-        return parsedScore;
+        return Integer.parseInt(trimmedMaxScore);
```

Comment:
This is a similar issue happening in gradeAssignmentCommand. Great attempt to simplify the code.